### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by dasch-1**

### CODE REVIEW

File: buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/ExtractedData.java [hunk 1]
```diff
@@ -33,9 +33,9 @@
...
 
   public ExtractedData() {
-    keywords = new LinkedHashMap<Keyword, String>();
-    nonReservedKeywords = new HashSet<Keyword>();
-    functions = new LinkedHashMap<String, String>();
-    tokenAssignments = new LinkedList<String>();
+    keywords = new LinkedHashMap<>();
+    nonReservedKeywords = new HashSet<>();
+    functions = new LinkedHashMap<>();
+    tokenAssignments = new ArrayList<>();
```
Comment (AndrewPochapsky) [inline buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/ExtractedData.java:39]: Just wondering, why the change to ArrayList throughout the code?

Comment (dasch-1) [reply buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/ExtractedData.java:39]: Our linter says it's more efficient than LinkedList.
